bin/install_cmdstan

```diff
@@ -0,0 +1,164 @@
+
+
+@contextlib.contextmanager
+def pushd(new_dir):
+    previous_dir = os.getcwd()
+    os.chdir(new_dir)
+    yield
+    os.chdir(previous_dir)
+
+
+def usage():
+    print(
+        """Arguments:
+        -v (--version) :CmdStan version
+        -d (--dir) : install directory
+        -h (--help) : this message
+        """)
+
+
+def install_version(cmdstan_version):
+    with pushd(cmdstan_version):
+        print('Building {} binaries'.format(cmdstan_version))
+        cmd = ['make', 'build']
+        proc = subprocess.Popen(
+            cmd, cwd=None, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        proc.wait()
+        stdout, stderr = proc.communicate()
+        if proc.returncode:
+            print('Command "make build" failed')
+            if stderr:
+                print(stderr.decode('ascii').strip())
+            sys.exit(3)
+        print('Test model compilation')
+        cmd = ['make', os.path.join('examples', 'bernoulli', 'bernoulli')]
+        proc = subprocess.Popen(
+            cmd, cwd=None, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        proc.wait()
+        stdout, stderr = proc.communicate()
+        if proc.returncode:
+            print('Failed to compile example model bernoulli.stan')
+            if stderr:
+                print(stderr.decode('ascii').strip())
+            sys.exit(3)
+    print('Installed {}'.format(cmdstan_version))
+
+
+def is_installed(cmdstan_version):
+    if not os.path.exists(os.path.join(cmdstan_version, 'bin')):
+        return False
+    if os.path.exists(os.path.join(
+            cmdstan_version, 'examples', 'bernoulli', 'bernoulli')):
+        return True
+    if os.path.exists(os.path.join(
+            cmdstan_version, 'examples', 'bernoulli', 'bernoulli.exe')):
+        return True
+    return False
+
+
+def latest_version():
+    try:
+        file_tmp, _ = urllib.request.urlretrieve(
+            'https://api.github.com/repos/stan-dev/cmdstan/releases/latest')
+    except urllib.error.URLError as err:
+        print('Cannot connect to github.')
+        print(err)
+        sys.exit(3)
+    with open(file_tmp, 'r') as fd:
+        response = fd.read()
+        start_idx = response.find('\"tag_name\":\"v') + len('"tag_name":"v')
+        end_idx = response.find('\"', start_idx)
+    return response[start_idx:end_idx]
+
+
+def retrieve_latest_version(version):
+    print('Downloading CmdStan version {}'.format(version))
+    url = 'https://github.com/stan-dev/cmdstan/releases/download/'\
+      'v{0}/cmdstan-{0}.tar.gz'.format(version)
+    try:
+        file_tmp, _ = urllib.request.urlretrieve(url, filename=None)
+    except urllib.error.URLError as err:
+        print('Failed to download CmdStan version {} from github.com'.format(
+            version))
+        print(err)
+        sys.exit(3)
+    print('Download successful, file: {}'.format(file_tmp))
+    try:
+        tar = tarfile.open(file_tmp)
+        tar.extractall(os.getcwd())
```

Comment:
Is there some progressbar option?

Reply:
progressbar option as part of `extractall` function?  no.

agreed that we need progress bars for download, untar, and compile - will file an issue.